### PR TITLE
feat(settings): add useAlertBar hook

### PR DESCRIPTION
**Pull request by jodyheavener**
## Because

We have a handful of hooks and functions that make up AlertBar usage

## This pull request

Consolidates the various hooks and functions we use with AlertBar into a single hook.

Now you can use the `useAlertBar` hook in combination with the `AlertBar` component to...

- Hide, show, and check the visibility of the alert bar
- Set the type and contents of the alert bar
- Use the `success`, `info`, and `error` convenience methods to do the above much quicker

## Issue that this pull request solves

Closes: #6316

## Checklist

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.

### CODE REVIEW

File: packages/fxa-settings/src/lib/hooks.tsx
Comment (dannycoates) [inline packages/fxa-settings/src/lib/hooks.tsx:100]: ReactNode includes both string and null in it's typedef

**Comment by jodyheavener** [comment]
I'll wait until #6351 is merged so I can update the docs to mention this hook and document its usage.

**Review by meandavejustice — APPROVED** [review]
Such a nice cleanup. I'm approving but sounds like you wanted to update the docs on this before merging.

**Comment by LZoog** [comment]
Heh, apparently my review was left "pending" from last week, but my comments are now out of date.

I did want to say - YAAAS this is a great improvement. Get those doc edits in since what we've got under AlertBar will now be out of date and then we'll be g2g. (If you merge before then... 
image)

**Pull request merged**
